File: C/majority_element.c
/* Write a program to find the majority element in the array. A majority element in an array A[] of size n is an element that appears more than n/2 
times (and hence there is at most one such element).  If input array doesn't contain a majority element, then output "NO Majority Element"

Input:  The first line of the input contains T denoting the number of testcases. The first line of the test case will be the size of array and 
second line will be the elements of the array.
Output: For each test case the output will be the majority element of the array.

Sample Input:
2
5
3 1 3 3 2
3
1 2 3

Sample Output:
3
NO Majority Element  */
#include <stdio.h>

int main() {
	int t;
	scanf("%d",&t);
	while(t>0){
	    int n;
	    scanf("%d",&n);
	    int a[n],b[101]={0},i,max=-1;
	    for ( i=0; i<n; i++ ) {
	        scanf("%d",&a[i]);
	        b[a[i]]+=1;
	    }
	    for(i=0;i<=100;i++){
	        if(b[i]>n/2)
	            max=i;
	    }
	    if(max>-1)
	        printf("%d\n",max);
	    else
	        printf("NO Majority Element\n");
	    t--;
	}
	return 0;
}

File: Python/GeeksforGeeks/majority-element.py
"""
Problem Link: https://practice.geeksforgeeks.org/problems/majority-element/0

Given an array A of N elements. Find the majority element in the array. 
A majority element in an array A of size N is an element that appears more than N/2 times in the array.

Input:  
The first line of the input contains T denoting the number of testcases. 
The first line of the test case will be the size of array and second line will be the elements of the array.

Output: 
For each test case the output will be the majority element of the array. 
Output "-1" if no majority element is there in the array.

Constraints:
1 <= T<= 100
1 <= N <= 107
0 <= Ai <= 106

Example:
Input:
2
5
3 1 3 3 2
3
1 2 3

Output:
3
-1

Explanation:
Testcase 1: Since, 3 is present more than N/2 times, so it is the majority element.
"""
for _ in range(int(input())):
    n = int(input())
    arr = list(map(int,input().split()))
    count = no = 0
    for i in arr:
        if count == 0:
            count += 1
            no = i
        elif no == i:
            count += 1
        else:
            count -= 1
    count = 0
    for i in arr:
        if i == no:
            count += 1
    print("-1" if count <= n/2 else no)